### PR TITLE
[SDP-1095] Add Authentication to GET `/organization/logo` 

### PR DESCRIPTION
**Pull request by marwen-abid**
### What
Adding authentication to `GET /organization/logo` 

### Why
Logos need to be able to resolve the tenant name. Making the endpoint authenticated solves this issue as we can reliably resolve the tenant id from the jwt token. 

FE PR: https://github.com/stellar/stellar-disbursement-platform-frontend/pull/79
Jira: https://stellarorg.atlassian.net/browse/SDP-1095

Screenshot 2024-03-04 at 11 54 52 AM
Screenshot 2024-03-04 at 11 55 00 AM

### Known limitations
n/a

#### PR Structure

* [x] This PR has a reasonably narrow scope (if not, break it down into smaller PRs).
* [x] This PR title and description are clear enough for anyone to review it.
* [x] This PR does not mix refactoring changes with feature changes (split into two PRs otherwise).

#### Thoroughness

* [x] This PR adds tests for the new functionality or fixes.
* [x] This PR contains the link to the Jira ticket it addresses.

#### Configs and Secrets

* [x] No new CONFIG variables are required -OR- the new required ones were added to the helmchart's [`values.yaml`] file.
* [x] No new CONFIG variables are required -OR- the new required ones were added to the deployments ([`pr-preview`], [`dev`], [`demo`], `prd`).
* [x] No new SECRETS variables are required -OR- the new required ones were mentioned in the helmchart's [`values.yaml`] file.
* [x] No new SECRETS variables are required -OR- the new required ones were added to the deployments ([`pr-preview secrets`], [`dev secrets`], [`demo secrets`], `prd secrets`).

#### Release

* [x] This is not a breaking change.
* [x] **This is ready for production.**. If your PR is not ready for production, please consider opening additional complementary PRs using this one as the base. Only merge this into `develop` or `main` after it's ready for production!

#### Deployment

* [ ] Does the deployment work after merging?

[`values.yaml`]: ../helmchart/sdp/values.yaml
[`pr-preview`]: https://github.com/stellar/kube/blob/d3e4f5dd8aa4c13b45a31a5a937f3e98841171a7/kube001-dev/namespaces/common-previews/stellar-disbursement-platform/backend-helm-values
[`dev`]: https://github.com/stellar/kube/blob/d3e4f5dd8aa4c13b45a31a5a937f3e98841171a7/kube001-dev/namespaces/stellar-disbursement-platform/backend-helm-values
[`demo`]: https://github.com/stellar/kube/blob/d3e4f5dd8aa4c13b45a31a5a937f3e98841171a7/kube001-dev/namespaces/stellar-disbursement-platform/demo/demo-backend-helm-values
[`pr-preview secrets`]: https://github.com/stellar/kube/blob/d3e4f5dd8aa4c13b45a31a5a937f3e98841171a7/kube001-dev/namespaces/common-previews/externalsecrets-common-previews.yaml#L241-L346
[`dev secrets`]: https://github.com/stellar/kube/blob/d3e4f5dd8aa4c13b45a31a5a937f3e98841171a7/kube001-dev/namespaces/stellar-disbursement-platform/stellar-disbursement-platform-externalsecrets.yaml
[`demo secrets`]: https://github.com/stellar/kube/blob/d3e4f5dd8aa4c13b45a31a5a937f3e98841171a7/kube001-dev/namespaces/stellar-disbursement-platform/demo/demo-sdp-externalsecrets.yaml

### CODE REVIEW

File: internal/serve/serve.go [hunk 1]
```diff
@@ -362,6 +363,9 @@ func handleHTTP(o ServeOptions) *chi.Mux {
 
 			r.With(middleware.AnyRoleMiddleware(authManager, data.GetAllRoles()...)).
 				Get("/", profileHandler.GetOrganizationInfo)
+
+			r.With(middleware.AnyRoleMiddleware(authManager, data.GetAllRoles()...)).
+				Get("/logo", profileHandler.GetOrganizationLogo)
```
Comment (marcelosalloum) [inline internal/serve/serve.go:368]: Can you also update the Test_handleHTTP_unauthenticatedEndpoints and Test_handleHTTP_authenticatedEndpoints tests accordingly to reflect this change?

File: internal/serve/serve.go [hunk 2]
```diff
@@ -344,6 +344,7 @@ func handleHTTP(o ServeOptions) *chi.Mux {
 			BaseURL:               o.BaseURL,
 			DistributionPublicKey: o.DistributionPublicKey,
 			PasswordValidator:     o.PasswordValidator,
+			PublicFilesFS:         publicfiles.PublicFiles,
```
Comment (marcelosalloum) [inline internal/serve/serve.go:347]: Noice

**Review by marcelosalloum — APPROVED** [review]
LGTM!

FYI: the deadcode check started failing b/c the latest version has changed. You can fix it by updating ci.yml in locking the versions of deadlock and exhaustive with this installation command(s):
```yml
          go install github.com/nishanths/exhaustive/cmd/exhaustive@v0.12.0
          go install golang.org/x/tools/cmd/deadcode@v0.18.0
```

File: internal/serve/serve.go (hunk 1 above)
Comment (marwen-abid) [reply internal/serve/serve.go:368]: Addressed in [d6af6f1](https://github.com/stellar/stellar-disbursement-platform-backend/pull/214/commits/d6af6f156af0be9a6a65b8dd72a75e3bbf80b69a)

**Pull request merged**